### DISCUSSION TITLE
Mobile Navbar in Nextjs/Tailwind isn't overlapping page content when opened

### DISCUSSION THREAD

**Discussion by github-ascnet** [tailwindlabs/tailwindcss#11803] [Help]
Here is the code of the responsive navbar. The mobile menu is controlled by a useState hook. The styling depends on whether the MenuOpen is true or not. The problem with the mobile menu is that it's always opened in the background (one layer below), so that the links are not visible/selectable.

`
"use client";

import Image from "next/image";
import Link from "next/link";
import logo from "../public/logo.png";
import { AiOutlineMenu, AiOutlineClose } from "react-icons/ai";
import { FaRegUser } from "react-icons/fa";
import { useState } from "react";

const NavbarPage = () =\> {
const \[menuOpen, setMenuOpen\] = useState(false);

const handleNav = () =\> {
setMenuOpen(!menuOpen);
};

return (
\<nav className="sticky w-full h-24 shadow-xl bg-navbar-dark "\>
\<div className="flex justify-between items-center h-full w-full px-10 2xl:px-16"\>
\<Link href="/"\>
\<Image
            src={logo}
            alt="logo"
            width="305"
            height="305"
            className="cursor-pointer"
            priority
          /\>
\</Link\>
\<div className="hidden md:flex"\>
<ul className="hidden md:flex">
\<Link href="/home"\>
<li className="ml-10 hover:border-b text-xl text-white flex">
Home
</li>
\</Link\>
\<Link href="/about"\>
<li className="ml-10 hover:border-b text-xl text-white">
About
</li>
\</Link\>
\<Link href="/contact"\>
<li className="ml-10 hover:border-b text-xl text-white">
Contact
</li>
\</Link\>
</ul>
\</div\>

        <div onClick={handleNav} className="md:hidden cursor-pointer pl-24">
          <AiOutlineMenu size={25} style={{ fill: "white" }} />
        </div>
      </div>
    
      <div
        className={
          menuOpen
            ? "fixed z-50 left-0 top-0 w-[65%] md:hidden h-screen bg-[#ecf0f3] p-10 ease-in duration-500"
            : "fixed left-[-100%] top-0 p-10 ease-in duration-500"
        }
      >
        <div className="flex w-full items-center justify-end">
          <div onClick={handleNav} className="cursor-pointer">
            <AiOutlineClose size={25} />
          </div>
        </div>
    
        <div className="flex-col py-4">
          <ul>
            <Link href="/home">
              <li
                onClick={() => setMenuOpen(false)}
                className="py-4 cursor-pointer"
              >
                Home
              </li>
            </Link>
            <Link href="/about">
              <li
                onClick={() => setMenuOpen(false)}
                className="py-4 cursor-pointer"
              >
                About
              </li>
            </Link>
            <Link href="/contact">
              <li
                onClick={() => setMenuOpen(false)}
                className="py-4 cursor-pointer"
              >
                Contact
              </li>
            </Link>
          </ul>
        </div>
      </div>
    </nav>

);
};

export default NavbarPage;
`

**Comment by wongjn** [comment]
Consider applying a positive `z-index` that is higher than other positioned elements within the same stacking context to the top-level `<nav>` element, such as `z-50`:

```diff
 return (
-  <nav className="sticky w-full h-24 shadow-xl bg-navbar-dark ">
+  <nav className="sticky w-full h-24 shadow-xl bg-navbar-dark z-50">
     <div className="flex justify-between items-center h-full w-full px-10 2xl:px-16">
```
